cmake/SetupCxxFlags.cmake

```diff
@@ -8,6 +8,16 @@ option(OVERRIDE_ARCH "The architecture to use. Default is native." OFF)
 option(ENABLE_SPECTRE_DEBUG "Enable ASSERTs and other SPECTRE_DEBUG options"
   OFF)
 
+# Work around boost not building with clang 15
+# (https://github.com/boostorg/functional/pull/21)
+if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "15.0.0")
+    set(CMAKE_CXX_FLAGS "-D_HAS_AUTO_PTR_ETC=0 ${CMAKE_CXX_FLAGS}")
+  elseif(CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "15.0.0")
+    set(CMAKE_CXX_FLAGS "-D_HAS_AUTO_PTR_ETC=0 ${CMAKE_CXX_FLAGS}")
+  endif()
+endif()
+
```

Comment:
I use if/elseif here, even though it duplicates a line, to avoid lines over 80 characters. I tried continuing the if line with \, but make complained

Reply:
I believe CMake ignores newlines outside of strings (except as generic whitespace), so you should be able to wrap it just like C++.

(To any other readers: The previous comment contains a backslash before the last comma, but GitHub isn't displaying it.)